### PR TITLE
net-p2p/i2p: version bump

### PR DESCRIPTION
**Pull request by tharvik**
Update i2p to 0.9.24 and removal of 0.9.18.

Gentoo Bug #573978 for the version bump.
- update to jdk:1.7 because it now require it
- avoid having a too big patch to handle on update by replacing parts by `sed`
- some cosmetic changes about quoting variables
- cetralize some variables about I2P install directory or user home
- remove 0.9.18 (it was already not keyworded)

### CODE REVIEW

**Comment by idella** [comment]
@monsieurp

File: net-p2p/i2p/files/i2p-0.9.24_initd [hunk 1]
```diff
@@ -0,0 +1,36 @@
+#!/sbin/runscript
+# Copyright 1999-2013 Gentoo Foundation
```
Comment (gktrk) [inline net-p2p/i2p/files/i2p-0.9.24_initd]: the copyright year needs updating

**Comment by gktrk** [comment]
The ebuild copyright year also needs updating.

You can use 'repoman full' before committing to help you catch these QA problems.

**Comment by monsieurp** [comment]
Indeed. It's no big deal though but please update the copyright year accordingly so that I can merge this PR.

**Comment by tharvik** [comment]
281b2c2 should do it then (just have to wait for QA checks to finish).

**Comment by monsieurp** [comment]
No worries, your last commit modifies dates so we can move forward and merge the PR. Thanks for the bump and PR! :smiley:

**Pull request merged**